packages/imperative/src/rest/src/session/AuthOrder.ts

```diff
@@ -0,0 +1,621 @@
+                            sessCfg._authCache.authTypeToRequestToken = SessConstants.AUTH_TYPE_CERT_PEM;
+                            sessTypeToUse = SessConstants.AUTH_TYPE_TOKEN;
+                        } else {
+                            sessTypeToUse = SessConstants.AUTH_TYPE_CERT_PEM;
+                        }
+                    }
+                    break;
+                case SessConstants.AUTH_TYPE_NONE:
+                    sessTypeToUse = SessConstants.AUTH_TYPE_NONE;
+                    break;
+                default:
+                    // authOrder was validated. A wrong value now is our programming error.
+                    errMsg = `sessCfg.authTypeOrder contains an invalid authentication = ${nextAuth}.`;
+                    Logger.getImperativeLogger().error(errMsg);
+                    throw new ImperativeError({ msg: errMsg });
+            }
+            if (sessTypeToUse !== null) {
+                // stop looking for auth types after we find the first one
+                break;
+            }
+        }
+
+        if (sessTypeToUse === null) {
+            // When no creds are in the session, record the auth type as none.
+            sessCfg.type = SessConstants.AUTH_TYPE_NONE;
+
+        } else {
+            sessCfg.type = sessTypeToUse;
+        }
+
+        // remove all extra auth creds from the session
+        AuthOrder.removeExtraCredsFromSess(sessCfg);
+        Logger.getImperativeLogger().debug("Ending sessCfg = " + Censor.censorSession(sessCfg));
+    }
+
+    // ***********************************************************************
+    /**
+     * Cache the authOrder property from the supplied cmdArgs. If no authOrder exists
+     * in cmdArgs, a default authOrder is created and cached.
+     *
+     * @param sessCfg - Modified.
+     *      A session configuration object into which we store the auth cache.
+     *
+     * @param cmdArgs - Input.
+     *      The set of arguments that the calling function is using.
+     */
+    private static cacheAuthOrder<SessCfgType extends ISession>(
+        sessCfg: SessCfgType,
+        cmdArgs: ICommandArguments
+    ): void {
+        // have we already cached an authOrder?
+        if (sessCfg.authTypeOrder.length > 0) {
+            // start over with an empty auth order.
+            sessCfg.authTypeOrder = [];
+        }
+
+        sessCfg._authCache.didUserSetAuthOrder = false;
+        if (cmdArgs.authOrder) {
+            if (typeof cmdArgs.authOrder === "string") {
+                if (cmdArgs.authOrder.length > 0) {
+                    // user supplied an authOrder
+                    // convert user's comma-separated string into an array of auth types, and remove whitespace
+                    const userAuthOrder = cmdArgs.authOrder.split(',');
+                    for (let nextUserAuth of userAuthOrder) {
+                        nextUserAuth = nextUserAuth.trim();
```

Comment:
This is one thing that Sonar cloud pointed out.

It recommends not reassigning loop-counters, which makes sense for for-loops in principle. However, this is a `for-of` and unless we are adding more stuff to `userAuthOrder` inside this loop, I don't believe there will be an issue here.

To address this, we could in theory trim the variable before it gets assigned, which is basically remapping the entire array. 
```
for (let nextUserAuth of userAuthOrder.map((v) => v.trim())) {
```

---

Again, I don't think that's necessary, thus I'm not requesting this to be changed : )

Reply:
I'm thinking that we can keep it as-is, especially because we do not use the array after the loop and the array is `string[]` (meaning that modifying each element won't affect any references to an object/class instance/etc)

Reply:
Since I had to make another change to respond to other comments, I will just add the map operation.